external/bootstrap.cmake

```diff
@@ -48,14 +53,39 @@ if(result)
     message(FATAL_ERROR "CMake step for Third Party builds failed: ${result}")
 endif()
 
-execute_process(COMMAND ${CMAKE_COMMAND} --build .
+if(MSVC)
+  # We need to build for both Release and Debug fpr MSVC because
+  # it will not re-run this build if the build_type changes in ide.
+  execute_process(COMMAND ${CMAKE_COMMAND} --build . --config Release
+                    WORKING_DIRECTORY ${REPOSITORY_DIR}/build/ThirdParty
+                    RESULT_VARIABLE result
+  #                    OUTPUT_QUIET      ### Disable this to debug external buiilds
+  )
+  if(result)
+    message(FATAL_ERROR "build step for MSVC Release Third Party builds failed: ${result}")
+  endif()
+  
+  execute_process(COMMAND ${CMAKE_COMMAND} --build . --config Debug
+                    WORKING_DIRECTORY ${REPOSITORY_DIR}/build/ThirdParty
+                    RESULT_VARIABLE result
+  #                    OUTPUT_QUIET      ### Disable this to debug external buiilds
+  )
+  if(result)
+    message(FATAL_ERROR "build step for MSVC Debug Third Party builds failed: ${result}")
+  endif()
+  
+else(MSVC)
```

Comment:
This is correct? Else with ()

Reply:
> This is correct? Else with ()

If there is an argument to the else( ) or endif( )  then CMake will confirm that it is the same as the if( ).

Reply:
did not know that, cool